code/game/gamemodes/wizard/wizard.dm

```diff
@@ -158,22 +158,30 @@
 	wizard_mob.gene_stability += DEFAULT_GENE_STABILITY //magic
 	return 1
 
+// Checks if the game should end due to all wizards and apprentices being dead, or MMI'd/Borged
 /datum/game_mode/wizard/check_finished()
 	var/wizards_alive = 0
 	var/traitors_alive = 0
+
+	// Wizards
 	for(var/datum/mind/wizard in wizards)
 		if(!istype(wizard.current,/mob/living/carbon))
 			continue
 		if(wizard.current.stat==DEAD)
 			continue
+		if(istype(wizard.current, /obj/item/mmi)) // wizard is in an MMI, don't count them as alive
+			continue
 		wizards_alive++
 
+	// Apprentices - classified as "traitors"
 	if(!wizards_alive)
 		for(var/datum/mind/traitor in traitors)
 			if(!istype(traitor.current,/mob/living/carbon))

```

Comment:
shouldn't this already catch the wizards in an MMI? since it's not a `/mob/living/carbon` but an `/obj/item/mmi`

Reply:
You'd think so but it doesn't count for some reason. Without these changes implemented, I did two tests. In each test there was one wizard and one apprentice. One where the wizard was in an MMI and the apprentice died, and another test where the apprentice was in the MMI and the wizard died. In both of the tests, the game continued. This is specifically for MMIs too, for borgs it didn't matter. For example if I made the wizard into a borg and killed the apprentice, the game ends and the same goes for the opposite situation

Reply:
No it doesn’t catch it because when they’re in an MMI they’re a /mob/living/carbon/brain

https://github.com/ParadiseSS13/Paradise/blob/f0abbd38d26105cd9b807864b786348ab7319b81/code/modules/mob/living/carbon/brain/MMI.dm#L13